plutus-core/untyped-plutus-core/testlib/Evaluation/Builtins/Definition.hs

```diff
@@ -1277,6 +1278,47 @@ test_Case =
                     Left _                        -> False
                     Right EvaluationFailure       -> 0 > scrut || scrut >= fromIntegral (length is)
                     Right (EvaluationSuccess res) -> res == mkConstant () (is !! fromIntegral scrut)
+        , QC.testProperty "List success" . QC.withMaxSuccess 99 $
+            \(scrut :: [Integer]) ->
+                let
+                  term :: Term TyName Name DefaultUni DefaultFun ()
+                  term = runQuote $ do
+                    x <- freshName "x"
+                    xs <- freshName "xs"
+                    let
+                      listElem = mkTyBuiltin @_ @Integer ()
+                      list = mkTyBuiltin @_ @[Integer] ()
+                    pure $
+                      kase ()
+                        listElem
+                        (mkConstant () scrut)
+                        [ mkConstant @Integer () 42
+                        , lamAbs () x listElem $ lamAbs () xs list $ var () x
+                        ]
+                  expected []      = 42
+                  expected (x:_xs) = x
+                in Right (EvaluationSuccess . mkConstant () $ expected scrut) QC.===
+                        typecheckEvaluateCekNoEmit def defaultBuiltinCostModelForTesting term
+        , QC.testProperty "List any" . QC.withMaxSuccess 99 $
+            \(scrut :: [Integer]) ->
+                let
+                  term :: Term TyName Name DefaultUni DefaultFun ()
+                  term = runQuote $ do
+                    x <- freshName "x"
+                    xs <- freshName "xs"
+                    let
+                      listElem = mkTyBuiltin @_ @Integer ()
+                      list = mkTyBuiltin @_ @[Integer] ()
+                    pure $
+                      kase ()
+                        listElem
+                        (mkConstant () scrut)
+                        [ lamAbs () x listElem $ lamAbs () xs list $ var () x
```

Comment:
This doesn't cover the case when there's more branches than expected.

It's also out of sync with the `Bool` tests, which have very similar constraints. I do think it's a good idea to have the two that you have there, but then we should also

1. add a test for when there's more branches than expected
2. structure the `Bool` tests the same way

I felt like `Bool any` was an OK way to merge what you have here and the too-many-branches test into a single test, but I never actually measured coverage, so if you want to split it into two, that's even better I guess.